lib/internal/queue.js

```diff
@@ -45,16 +47,14 @@ export default function queue(worker, concurrency, payload) {
     }
 
     function _next(tasks) {
-        return function(){
+        return rest(function(args){
             workers -= 1;
 
-            var removed = false;
-            var args = arguments;
             arrayEach(tasks, function (task) {
-                arrayEach(workersList, function (worker, index) {
-                    if (worker === task && !removed) {
+                arraySome(workersList, function (worker, index) {
```

Comment:
currently a task can be called multiple times because we call all tasks I think.

Also yes on arrayEach, why?